### ISSUE TITLE
Cannot fetch Slack messages (due to unsupported enum value serialization in URI)

### ISSUE THREAD

**Issue by Waldo000000** [soxtoby/SlackNet#135]
**Repro**
```
var api = new SlackServiceBuilder()
	.UseApiToken("xoxp-your-user-token")
	.GetApiClient();
var messages = api
	.Search.Messages("from:<@YourUserId>")
	.Result;
```

**Expected**: 
1. Request should be placed to `https://slack.com/api/search.messages?...sort=score&sort_dir=desc...`
2. Response should contain Slack messages

**Actual**:

1. Request is placed to `https://slack.com/api/search.messages?...sort="score"&sort_dir="desc"...` (_note that serialized `sort` and `sort_dir` enum values are enclosed in quotes_)
2. Response is:
```
{
    "ok": false,
    "error": "invalid_arguments",
    "response_metadata": {
        "messages": [
            "[ERROR] must be a valid enum value [json-pointer:/sort]",
            "[ERROR] must be a valid enum value [json-pointer:/sort_dir]"
        ]
    }
}
```

**Comment by soxtoby** [comment]
Thanks for raising this, @Waldo000000. I'll have a fix pushed up shortly.

**Comment by soxtoby** [comment]
Should be fixed in v0.10.18. Turned out the messages in the search results didn't quite match `MessageEvent`, so they've been changed to `MessageSearchResult`s instead. The doco's not clear about what exactly is included in the message results objects, so I've included an `ExtraProperties` property in case I missed anything.
